include/seqan3/range/decorator/gap_decorator.hpp

```diff
@@ -1038,6 +656,389 @@ template <std::ranges::view urng_t>
+    }
+
+    //!\brief Return underlying container value currently pointed at.
+    constexpr reference operator[](difference_type const n) const noexcept
+    {
+        return *(*this + n);
+    }
+    //!\}
+
+    /*!\name Comparison operators
+     * \brief Compares iterators by virtual position.
+     * \{
+     */
+
+    //!\brief Checks whether `*this` is equal to `rhs`.
+    constexpr friend bool operator==(gap_decorator_iterator_ra const & lhs, gap_decorator_iterator_ra const & rhs)
+        noexcept
+    {
+        return lhs.pos == rhs.pos;
+    }
+
+    //!\brief Checks whether `*this` is not equal to `rhs`.
+    constexpr friend bool operator!=(gap_decorator_iterator_ra const & lhs, gap_decorator_iterator_ra const & rhs)
+        noexcept
+    {
+        return lhs.pos != rhs.pos;
+    }
+
+    //!\brief Checks whether `*this` is less than `rhs`.
+    constexpr friend bool operator<(gap_decorator_iterator_ra const & lhs, gap_decorator_iterator_ra const & rhs)
+        noexcept
+    {
+        return lhs.pos < rhs.pos;
+    }
+
+    //!\brief Checks whether `*this` is greater than `rhs`.
+    constexpr friend bool operator>(gap_decorator_iterator_ra const & lhs, gap_decorator_iterator_ra const & rhs)
+        noexcept
+    {
+        return lhs.pos > rhs.pos;
+    }
+
+    //!\brief Checks whether `*this` is less than or equal to `rhs`.
+    constexpr friend bool operator<=(gap_decorator_iterator_ra const & lhs, gap_decorator_iterator_ra const & rhs)
+        noexcept
+    {
+        return lhs.pos <= rhs.pos;
+    }
+
+    //!\brief Checks whether `*this` is greater than or equal to `rhs`.
+    constexpr friend bool operator>=(gap_decorator_iterator_ra const & lhs, gap_decorator_iterator_ra const & rhs)
+        noexcept
+    {
+        return lhs.pos >= rhs.pos;
+    }
+    //!\}
+};
+
+/*!\brief The iterator type over a seqan3::gap_decorator.
+ *
+ * \details
+ *
+ * This iterator returns values when dereferenced, not references, i.e. it does not satisfy the semantic
+ * requirements of [Cpp17BidirectionalIterator](https://en.cppreference.com/w/cpp/named_req/bidirectional_iterator).
+ * It does model the C++20 std::bidirectional_iterator. In addition, it offers all interfaces of a standard
+ * std::random_access_iterator except the iterator category which is std::bidirectional_iterator_tag, because the
+ * complexity of the iterator is logarithmic and not constant. However, all interfaces inside the seqan3::gap_decorator
+ * make use of the more efficient logarithmic implementation. Only if you want to use the seqan3::gap_decorator in a
+ * generic algorithm, e.g. std::ranges::distance, the slower linear version will be picked due to the constraints of
+ * the iterator category. To achieve optimal performance in a generic context you can use
+ * seqan3::views::pseudo_random_access to get a subrange over seqan3::gap_decorator::gap_decorator_iterator_ra which
```

Comment:
```suggestion
 * seqan3::views::pseudo_random_access to get a range over seqan3::gap_decorator::gap_decorator_iterator_ra which
```
I know it is of type subrange, but this sounds as if I can additionally crop the range.